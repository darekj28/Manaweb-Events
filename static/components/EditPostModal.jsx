var React = require('react');
// var $ = require('jquery');
export default class EditPostModal extends React.Component {
	constructor(props) {
		super(props);
		this.state = {postContent : this.props.post.postContent};
		this.handlePostEditChange = this.handlePostEditChange.bind(this);
		this.handlePostEditSubmit = this.handlePostEditSubmit.bind(this);
	}
	handlePostEditChange(e) {
		this.setState({ postContent : e.target.value});
	}
	handlePostEditSubmit() {
		var obj = {unique_id : this.props.post.unique_id,
						field_name : "body", 
						field_data : this.state.postContent};

		$.ajax({
			type : 'POST',
			url  : '/editPost',
			data : JSON.stringify(obj, null, '\t'),
			contentType: 'application/json;charset=UTF-8'
		});
		this.props.refreshFeed();
	}
	componentWillReceiveProps(nextProps) {
		this.setState({ postContent : nextProps.post.postContent });
	}
	render() {
		return(
			<div className="modal fade" id="EditPostModal" aria-hidden="true">
				<div className="modal-dialog" role="document">
					<div className="modal-content">
						<div className="modal-header">
							<button type="button" className="close" data-dismiss="modal" aria-label="close">
								<span aria-hidden="true">&times;</span>
							</button>
							<h4 className="modal-title">Edit Post</h4>
						</div>
						<div className="modal-body" id="EditPostModalBody">
							<textarea className="form-control EditFeedPostBody" id="FeedPostBody" 
									value={this.state.postContent} 
									onChange={this.handlePostEditChange}></textarea>
						</div>
						<div className="modal-footer">
							<button id="epm_submit" type="button" className="btn btn-default" data-dismiss="modal" 
									onClick={this.handlePostEditSubmit}>Submit</button>
						</div>
					</div>
				</div>
			</div>
		);
	}
}