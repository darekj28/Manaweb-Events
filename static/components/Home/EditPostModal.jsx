var React = require('react');
export default class EditPostModal extends React.Component {
	constructor(props) {
		super(props);
		this.state = {postContent : this.props.post.postContent};
	}
	handlePostEditChange(e) {
		this.setState({ postContent : e.target.value});
	}
	handlePostEditSubmit() {
		var obj = {unique_id : this.props.post.unique_id,
						field_name : "body", 
						field_data : this.state.postContent};
		
		$.ajax({
			type : 'POST',
			url  : '/editPost',
			data : JSON.stringify(obj, null, '\t'),
			contentType: 'application/json;charset=UTF-8'
		});
		this.props.handlePostEdit(this.props.post, this.state.postContent);
	}
	componentWillReceiveProps(nextProps) {
		this.setState({ postContent : nextProps.post.postContent });
	}
	render() {
		return(
			<div className="modal fade" id="EditPostModal" aria-hidden="true">
				<div className="modal-dialog" role="document">
					<div className="modal-content">
						<div className="modal-header">
							<button type="button" className="close" data-dismiss="modal" aria-label="close">
								<span aria-hidden="true">&times;</span>
							</button>
							<h4 className="modal-title">Edit Post</h4>
						</div>
						<div className="modal-body" id="EditPostModalBody">
							<textarea className="form-control EditFeedPostBody" id="FeedPostBody" 
									value={this.state.postContent} 
									onChange={this.handlePostEditChange.bind(this)}></textarea>
						</div>
						<div className="modal-footer">
							<button id="epm_submit" type="button" className="btn btn-default" data-dismiss="modal" 
									onClick={this.handlePostEditSubmit.bind(this)}>Submit</button>
						</div>
					</div>
				</div>
			</div>
		);
	}
}