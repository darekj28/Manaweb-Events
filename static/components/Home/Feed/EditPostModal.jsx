var React = require('react');
import AppStore from '../../../stores/AppStore.jsx';

export default class EditPostModal extends React.Component {
	constructor(props) {
		super(props);
		this.state = {postContent : this.props.post.postContent};
	}
	handlePostEditChange(e) {
		this.setState({ postContent : e.target.value});
	}
	handlePostEditSubmit() {
		var obj = {unique_id : this.props.post.unique_id,
					userID : AppStore.getCurrentUser()['userID'],
						field_name : "body", 
						field_data : this.state.postContent,
						jwt : localStorage.jwt};
		
		$.ajax({
			type : 'POST',
			url  : '/editPost',
			data : JSON.stringify(obj, null, '\t'),
			contentType: 'application/json;charset=UTF-8',
			success : function(data) {
				if (data['result'] == 'success')
					swal({title : "Success!", 
						text: "The post has been edited.", 
						type: "success",
						confirmButtonColor: "#80CCEE",
							confirmButtonText: "OK"
					});
				else 
					swal({title : "Sorry!", 
						text: "This post was deleted.", 
						type: "error",
						confirmButtonColor: "#80CCEE",
							confirmButtonText: "OK"
					});
			},
			error : function(data) {
				swal({title : "Sorry!", 
					text: "This post was deleted.", 
					type: "error",
					confirmButtonColor: "#80CCEE",
						confirmButtonText: "OK"
				});
			}
		});
		if (this.props.handlePostEdit)
			this.props.handlePostEdit(this.props.post, this.state.postContent);
	}
	componentWillReceiveProps(nextProps) {
		this.setState({ postContent : nextProps.post.postContent });
	}
	render() {
		return(
			<div className="modal fade" id="EditPostModal" aria-hidden="true">
				<div className="modal-dialog" role="document">
					<div className="modal-content">
						<div className="modal-header">
							<button type="button" className="close" data-dismiss="modal" aria-label="close">
								<span aria-hidden="true">&times;</span>
							</button>
							<div className="modal-title">Edit Post</div>
						</div>
						<div className="modal-body" id="EditPostModalBody">
							<textarea className="form-control edit-post" id="FeedPostBody" 
									value={this.state.postContent} 
									onChange={this.handlePostEditChange.bind(this)}></textarea>
						</div>
						<div className="modal-footer">
							<button id="epm_submit" type="button" className="btn post-button" data-dismiss="modal" 
									onClick={this.handlePostEditSubmit.bind(this)}>Submit</button>
						</div>
					</div>
				</div>
			</div>
		);
	}
}