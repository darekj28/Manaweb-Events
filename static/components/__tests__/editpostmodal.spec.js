import EditPostModal from '../EditPostModal.jsx';

describe('EditPostModal', () => {
	it('should call handlePostEditSubmit when submit is clicked', () => {
		window.$ = {ajax: jest.fn()}
		const refreshFeed = jest.fn();
		const post = {unique_id : 1};
		const wrapper = mount(
			<EditPostModal post={post} refreshFeed={refreshFeed}/>
			);
		const btn = wrapper.find('#epm_submit');
		btn.simulate('click');
		expect(refreshFeed).toHaveBeenCalled();
	});
	it('renders correctly', () => {
		const refreshFeed = jest.fn();
		const post = {unique_id : 1};
		const epm = renderer.create(
			<EditPostModal post={post} refreshFeed={refreshFeed}/>
			);
		expect(epm).toMatchSnapshot();
	});

})