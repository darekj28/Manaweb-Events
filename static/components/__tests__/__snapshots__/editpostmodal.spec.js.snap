exports[`EditPostModal renders correctly 1`] = `
<div
  aria-hidden="true"
  className="modal fade"
  id="EditPostModal">
  <div
    className="modal-dialog"
    role="document">
    <div
      className="modal-content">
      <div
        className="modal-header">
        <button
          aria-label="close"
          className="close"
          data-dismiss="modal"
          type="button">
          <span
            aria-hidden="true">
            Ã—
          </span>
        </button>
        <h4
          className="modal-title">
          Edit Post
        </h4>
      </div>
      <div
        className="modal-body"
        id="EditPostModalBody">
        <textarea
          className="form-control EditFeedPostBody"
          id="FeedPostBody"
          onChange={[Function]}
          value={undefined} />
      </div>
      <div
        className="modal-footer">
        <button
          className="btn btn-default"
          data-dismiss="modal"
          id="epm_submit"
          onClick={[Function]}
          type="button">
          Submit
        </button>
      </div>
    </div>
  </div>
</div>
`;
